sdk/database/helper/connutil/sql.go

```diff
@@ -29,8 +41,15 @@ type SQLConnectionProducer struct {
 	MaxConnectionLifetimeRaw interface{} `json:"max_connection_lifetime" mapstructure:"max_connection_lifetime" structs:"max_connection_lifetime"`
 	Username                 string      `json:"username" mapstructure:"username" structs:"username"`
 	Password                 string      `json:"password" mapstructure:"password" structs:"password"`
+	AuthType                 string      `json:"auth_type" mapstructure:"auth_type" structs:"auth_type"`
+	Credentials              string      `json:"credentials" mapstructure:"credentials" structs:"credentials"`
```

Comment:
Do we see `Credentials` is being a generic field which other types of credentials could be delivered through in the future? The naming is generic although it is being used for something specific right now (GCP service account credentials JSON).

If we do envision this field being used for other credentials in the future, I wonder if the `string` type will be supportive of this. For example, if we needed to support something like AWS IAM credentials for a database, what could this potentially look like?

If the `Credentials` field alone wouldn't support future use cases, should the naming be more specific (something like service_account_json)?

Reply:
That's a great point! We had a similar thought when chatting about this and appreciate the feedback. We were not thinking that this would be a generic field. For other services like AWS IAM, adding separate parameters like `aws_access_key_id` and `aws_secret_access_key` was what we had in mind.

I like renaming the field to be GCP specific, such as `service_account_json`, and will make that change!

Reply:
Updated the field to `service_account_json` be more explicit